src/depackers/arcfs.c

```diff
@@ -27,10 +27,10 @@ struct archived_file_header_tag {
 	unsigned char method;
 	unsigned char bits;
 	char name[13];
-	unsigned long compressed_size;
+	unsigned int compressed_size;
 	unsigned int date, time, crc;
-	unsigned long orig_size;
-	unsigned long offset;
+	unsigned int orig_size;
+	unsigned int offset;
```

Comment:
Considering some of these fields were `long` the `int`s are suspect too—are those supposed to be 16-bit DOS ints? I think using `fread` hacks to fill in struct data like this should be avoided for exactly this sort of issue. Like you mentioned, stdint.h or compatible types could also be used instead... probably better to just get rid of the offending `fread` usage though. Other bad things here: there's an implicit padding byte after `name` that is hidden by this hack.

Reply:
The `int`s are most probably amiga ints and 32 bits.